### ISSUE TITLE
fix(core): LModule — move grad-norm monitoring to on_before_optimizer_step

### ISSUE THREAD

**Issue by CedrickArmel** [CedrickArmel/radiologist#61]
## Context / background

`LModule.configure_gradient_clipping` (`radiologist-core/src/radiologist/core/module.py`) currently logs `grad_norm` and `weight_norm_post_clip` inside the clipping hook. Two bugs:

1. **`grad_norm` is silently inactive when no `gradient_clip_val` is configured.** Lightning only invokes `configure_gradient_clipping` when gradient clipping is enabled, so with clipping off (the common case) the pre-clip gradient norm is never logged.
2. **`weight_norm_post_clip` duplicates `weight_norm`.** `on_train_batch_end` already logs `weight_norm`, and clipping does not modify weights (only gradients), so `weight_norm_post_clip` is redundant.

## Motivation

Gradient-norm monitoring must be always-on (it is a primary training-health signal), independent of whether clipping is enabled. Post-clip gradient norm is only meaningful when clipping actually runs, so it stays in the clipping hook. Remove the redundant `weight_norm_post_clip`.

## Goals

- Add `on_before_optimizer_step(optimizer)` that logs `grad_norm` (pre-clip) — **always fires**, every optimizer step.
- In `configure_gradient_clipping`: clip first, then log `grad_norm_post_clip` — fires **only** when clipping is enabled.
- Remove the `weight_norm_post_clip` log.
- Keep `weight_norm` logging in `on_train_batch_end` unchanged.

## Expected vs actual behavior

**Actual:** `grad_norm` is only logged when `gradient_clip_val` is set; `weight_norm_post_clip` duplicates `weight_norm`.

**Expected:** `grad_norm` is logged on every optimizer step regardless of clipping config; `grad_norm_post_clip` is logged only when clipping runs; `weight_norm_post_clip` no longer exists.

## Acceptance criteria

- [ ] Running a training step with **no** gradient clipping configured logs a `grad_norm` metric (via `on_before_optimizer_step`).
- [ ] Running a training step **with** gradient clipping configured logs both `grad_norm` (pre-clip) and `grad_norm_post_clip` (post-clip), and the post-clip value is ≤ the configured clip threshold for norm-based clipping.
- [ ] No `weight_norm_post_clip` metric is ever logged.
- [ ] `weight_norm` continues to be logged once per training batch via `on_train_batch_end`.
- [ ] mypy clean; pytest green.

## Design / interface contracts

```python
def on_before_optimizer_step(self, optimizer: Any) -> None:
    """Log pre-clip gradient L2 norm. Fires every optimizer step."""
    grad_norm = self._get_grad_norm()
    self.log("grad_norm", grad_norm, on_step=True, prog_bar=False)

def configure_gradient_clipping(
    self,
    optimizer: Any,
    gradient_clip_val: Optional[float] = None,
    gradient_clip_algorithm: Optional[str] = None,
) -> None:
    """Clip gradients, then log post-clip grad norm.

    Only invoked by Lightning when clipping is enabled, so grad_norm_post_clip
    is logged exactly when clipping actually runs.
    """
    self.clip_gradients(
        optimizer,
        gradient_clip_val=gradient_clip_val,
        gradient_clip_algorithm=gradient_clip_algorithm,
    )
    grad_norm_post_clip = self._get_grad_norm()
    self.log("grad_norm_post_clip", grad_norm_post_clip, on_step=True, prog_bar=False)
```

`on_train_batch_end` is unchanged:

```python
def on_train_batch_end(self, outputs: Any, batch: Any, batch_idx: int) -> None:
    weight_norm = self._get_weight_norm()
    self.log("weight_norm", weight_norm, on_step=True, prog_bar=False)
```

## Technical notes

- File: `radiologist-core/src/radiologist/core/module.py`.
- `_get_grad_norm()` and `_get_weight_norm()` already exist and are reused as-is.
- Lightning calls `on_before_optimizer_step` on every optimizer step regardless of clipping config — this is what makes `grad_norm` always-on. It calls `configure_gradient_clipping` only when clipping is enabled.

## Out of scope

- Changes to `_get_grad_norm` / `_get_weight_norm` implementations.
- Setting clip values in Hydra configs (this issue only changes where/when norms are logged).
- The broader LModule overhaul — see #60.

## Dependencies

- Related: #60 — both edit `module.py`; coordinate merge order.
